### PR TITLE
[REF] product_customer_code: migrate to new api

### PR DESCRIPTION
**Pull request by jorgescalona**
HU#1634
VX#8010

It is required to migrate the module to new api

TODO:

- [X] Migrate imports
- [X] Migrate classes and methods
- [X] functional testing
- [X] Make dummy

### CODE REVIEW

**Comment by jorgescalona** [comment]
@JulioSerna is ready to check pls!

File: product_customer_code/product.py [hunk 1]
```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
...
-                    ids.append(ppu.product_id.id)
-            if ids:
-                res = self.name_get(cr, user, ids, context)
+                id_prod_code = product_customer_code_obj.search(
+                    [('product_code', '=', name),
+                     ('partner_id', '=', partner_id)], limit=limit,)
+                id_prod = id_prod_code and (
+                    product_customer_code_obj.browse(
+                        id_prod_code, context=self._context) or [])
+            if id_prod:
```
Comment (moylop260) [inline product_customer_code/product.py]: Possible undefined variable `id_prod` if the previous conditions are not executed

File: product_customer_code/product.py [hunk 2]
```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
...
-    def name_search(self, cr, user, name='', args=None, operator='ilike',
-                    context=None, limit=80):
-        res = super(ProductProduct, self).name_search(
-            cr, user, name, args, operator, context, limit)
-        if not context:
-            context = {}
-        product_customer_code_obj = self.pool.get('product.customer.code')
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=80):
+        if args is None:
```
Comment (moylop260) [inline product_customer_code/product.py]: You are not using `args` so you don't need re-defined here

File: product_customer_code/product.py [hunk 3]
```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
...
-                for ppu in id_prod:
-                    ids.append(ppu.product_id.id)
-            if ids:
-                res = self.name_get(cr, user, ids, context)
+                id_prod_code = product_customer_code_obj.search(
+                    [('product_code', '=', name),
+                     ('partner_id', '=', partner_id)], limit=limit,)
+                id_prod = id_prod_code and (
+                    product_customer_code_obj.browse(
+                        id_prod_code, context=self._context) or [])
```
Comment (moylop260) [inline product_customer_code/product.py]: Is required send `self._context` explicitly for 8.0 yet?

File: product_customer_code/product.py [hunk 4]
```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
...
+            if id_prod:
+                res = id_prod.name_get()
```
Comment (moylop260) [inline product_customer_code/product.py]: You don't need `name_get` of `product_customer_code_obj` you need `name_get` of `product.product`

File: product_customer_code/product_customer_code.py [hunk 5]
```diff
@@ -23,44 +23,44 @@
...
+
+    @api.model
+    def default_get(self, default_fields):
+        rec = super(ProductCustomerCode, self).default_get(default_fields)
+        rec['company_id'] = (
+            self.env['res.company']._company_default_get(
+                'product.customer.code'))
+        return rec
+
+    company_id = fields.Many2one('res.company', 'Company',
```
Comment (moylop260) [inline product_customer_code/product_customer_code.py]: Why you are not using the default parameter?

File: product_customer_code/product_customer_code.py [hunk 6]
```diff
@@ -23,44 +23,44 @@
...
+                               quotation.""")
+
+    product_id = fields.Many2one('product.product', 'Product', required=True,
+                                 help='Product Identification')
+
+    partner_id = fields.Many2one('res.partner', 'Customer', required=True,
+                                 help='Partner Reference')
+
+    @api.model
+    def default_get(self, default_fields):
```
Comment (moylop260) [inline product_customer_code/product_customer_code.py]: why?
 - fields definition
 - methods
 - fields definition

Use:
 - fields definition (all)
 - methods

File: product_customer_code/product_customer_code.py [hunk 7]
```diff
@@ -23,44 +23,44 @@
...
+    company_id = fields.Many2one('res.company', 'Company',
+                                 help='Company Reference',
+                                 )
 
     _sql_constraints = [
         ('unique_code', 'unique(product_code,company_id,partner_id)',
          'Product Code of customer must be unique'),
     ]
-
-    # TODO: Add index to product_code, partner_id
```
Comment (moylop260) [inline product_customer_code/product_customer_code.py]: You are removing the `TODO` but I didn't see the `index` parameter for these fields

File: product_customer_code/product.py [hunk 8]
```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
...
-                id_prod = id_prod_code and product_customer_code_obj.browse(
-                    cr, user, id_prod_code, context=context) or []
-                for ppu in id_prod:
-                    ids.append(ppu.product_id.id)
-            if ids:
-                res = self.name_get(cr, user, ids, context)
+                id_prod_code = product_customer_code_obj.search(
+                    [('product_code', '=', name),
+                     ('partner_id', '=', partner_id)], limit=limit,)
+                id_prod = id_prod_code and (
```
Comment (moylop260) [inline product_customer_code/product.py]: You don't need this `value and browse or []` for new browse of 8.0

File: product_customer_code/product.py [hunk 9]
```diff
@@ -23,18 +23,15 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
```
Comment (moylop260) [inline product_customer_code/product.py]: Use short license

File: product_customer_code/product.py [hunk 10]
```diff
@@ -23,18 +23,15 @@
...
 ##############################################################################
-from openerp import api
-from openerp.osv import osv, fields
+from openerp import models, fields, api
```
Comment (moylop260) [inline product_customer_code/product.py]: use isort

**Comment by moylop260** [comment]
I didn't see a video url to see it working functionally

File: product_customer_code/product.py [hunk 11]
```diff
@@ -23,18 +23,15 @@
...
     _inherit = "product.product"
 
-    _columns = {
-        'product_customer_code_ids': fields.one2many('product.customer.code',
-                                                     'product_id',
-                                                     'Customer Codes'),
-    }
+    product_customer_code_ids = fields.One2many('product.customer.code',
+                                                'product_id',
+                                                help='Customer Codes')
```
Comment (moylop260) [inline product_customer_code/product.py]: The string of this field is missing

File: product_customer_code/product.py (hunk 4 above)
Comment (moylop260) [inline product_customer_code/product.py]: If we will overwrite the original super value so what is the reason to call the super method and re-compute processing to do nothing

File: product_customer_code/product_customer_code.py [hunk 12]
```diff
@@ -23,44 +23,44 @@
...
+    product_id = fields.Many2one('product.product', 'Product', required=True,
+                                 help='Product Identification')
+
+    partner_id = fields.Many2one('res.partner', 'Customer', required=True,
+                                 help='Partner Reference')
+
+    @api.model
+    def default_get(self, default_fields):
+        rec = super(ProductCustomerCode, self).default_get(default_fields)
+        rec['company_id'] = (
```
Comment (moylop260) [inline product_customer_code/product_customer_code.py]: How you avoid index error here?

**Comment by JulioSerna** [comment]
replace by PR https://github.com/Vauxoo/addons-vauxoo/pull/1206

**Pull request closed without merging**